internal/api/v1/webhook.go

```diff
@@ -77,3 +89,1208 @@ func (h *WebhookHandler) GetDashboardURL(c *gin.Context) {
+		return
+	}
+
+	// Get Stripe signature from headers
+	signature := c.GetHeader("Stripe-Signature")
+	if signature == "" {
+		h.logger.Errorw("missing Stripe-Signature header")
+		c.JSON(http.StatusBadRequest, gin.H{
+			"error": "Missing Stripe-Signature header",
+		})
+		return
+	}
+
+	// Set context with tenant and environment IDs
+	ctx := types.SetTenantID(c.Request.Context(), tenantID)
+	ctx = types.SetEnvironmentID(ctx, environmentID)
+	c.Request = c.Request.WithContext(ctx)
+
+	// Get Stripe connection to retrieve webhook secret
+	conn, err := h.stripeService.ConnectionRepo.GetByEnvironmentAndProvider(ctx, environmentID, types.SecretProviderStripe)
+	if err != nil {
+		h.logger.Errorw("failed to get Stripe connection for webhook verification",
+			"error", err,
+			"environment_id", environmentID,
+		)
+		c.JSON(http.StatusBadRequest, gin.H{
+			"error": "Stripe connection not configured for this environment",
+		})
+		return
+	}
+
+	// Get Stripe configuration including webhook secret
+	stripeConfig, err := h.stripeService.GetDecryptedStripeConfig(conn)
+	if err != nil {
+		h.logger.Errorw("failed to get Stripe configuration",
+			"error", err,
+			"environment_id", environmentID,
+		)
+		c.JSON(http.StatusBadRequest, gin.H{
+			"error": "Invalid Stripe configuration",
+		})
+		return
+	}
+
+	// Verify webhook secret is configured
+	if stripeConfig.WebhookSecret == "" {
+		h.logger.Errorw("webhook secret not configured for Stripe connection",
+			"environment_id", environmentID,
+		)
+		c.JSON(http.StatusBadRequest, gin.H{
+			"error": "Webhook secret not configured",
+		})
+		return
+	}
+
+	// Log webhook processing (without sensitive data)
+	h.logger.Debugw("processing webhook",
+		"environment_id", environmentID,
+		"tenant_id", tenantID,
+		"payload_length", len(body),
+	)
+
+	// Parse and verify the webhook event
+	event, err := h.stripeService.ParseWebhookEvent(body, signature, stripeConfig.WebhookSecret)
+	if err != nil {
+		h.logger.Errorw("failed to parse/verify Stripe webhook event", "error", err)
+		c.JSON(http.StatusBadRequest, gin.H{
+			"error": "Failed to verify webhook signature",
+		})
+		return
+	}
+
+	// Handle different event types
+	switch string(event.Type) {
+	case string(types.WebhookEventTypeCustomerCreated):
+		h.handleCustomerCreated(c, event, environmentID)
+	case string(types.WebhookEventTypeCheckoutSessionCompleted):
+		h.handleCheckoutSessionCompleted(c, event, environmentID)
+	case string(types.WebhookEventTypeCheckoutSessionAsyncPaymentSucceeded):
+		h.handleCheckoutSessionAsyncPaymentSucceeded(c, event, environmentID)
+	case string(types.WebhookEventTypeCheckoutSessionAsyncPaymentFailed):
+		h.handleCheckoutSessionAsyncPaymentFailed(c, event, environmentID)
+	case string(types.WebhookEventTypeCheckoutSessionExpired):
+		h.handleCheckoutSessionExpired(c, event, environmentID)
+	case string(types.WebhookEventTypePaymentIntentPaymentFailed):
+		h.handlePaymentIntentPaymentFailed(c, event, environmentID)
```

Comment:
all these functions are stripe specific and hence can be in either stipe service or a separate stripe package (preferred)

Also we can pass on the connection object in these function so that we are not fetching and decrypting the stripe object everywhere in each function